mesa/space.py

```diff
@@ -34,12 +37,44 @@ class Grid(object):
         get_cell_list_contents: Returns the contents of a list of cells ((x,y) tuples)
     '''
 
-    width = None
-    height = None
-    torus = False
-    grid = None
     default_val = lambda s: None
 
+
+    class GridOccupiedIter:
```

Comment:
class GridOccupiedCells

That might work.

The whole idea of using iterators is to make it easier for novice programmers: instead of nested loops over y and x, you just call an iterator.

But I am happy to give it a better name!


On Mar 10, 2015, at 11:15 PM, Jackie Kazil <notifications@github.com> wrote:

> In mesa/space.py:
> 
> >      default_val = lambda s: None
> >  
> > +
> > +    class GridOccupiedIter:
> @gcallah - this is a much nicer way of doing this. 
> GridOccupiedIter is not intuitive for non-programmers to use. Without understanding Python, you might not know what that is. Suggestions for another names? 
> @dmasad - do you have any thoughts on this?
> 
> —
> Reply to this email directly or view it on GitHub.
>